lib/defaults/defaults.go

```diff
@@ -762,23 +751,33 @@ var (
 
 	// FIPSCiphers is a list of supported FIPS compliant SSH ciphers.
 	FIPSCiphers = []string{
-		"aes128-ctr",
-		"aes192-ctr",
+		"aes256-gcm@openssh.com",
 		"aes256-ctr",
+		"aes192-ctr",
 		"aes128-gcm@openssh.com",
+		"aes128-ctr",
 	}
```

Comment:
If this is used by the client side (and if it isn't, why?) then the ordering will also dictate the selected cipher, as per RFC 4253 § 7.1:
> The chosen encryption algorithm to each direction MUST be the first algorithm on the client's name-list that is also on the server's name-list.

The previous ordering matched the ordering that OpenSSH uses, filtered for FIPS: `chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com`, which I assume was the intent.

Reply:
Good catch, @espadolini. I'll comment on the upstream PR as well to get them to align in the same way. I've added comments to each of the slices.

Reply:
We're not forced to follow what OpenSSH does, we could also pick the default ordering from `x/crypto/ssh`, we should just document why we picked the order we picked.

Reply:
Personally, I would go with GCM over CTR mode. GCM does both encryption and authentication in the same operation instead of separate (which only leads to problems). Since you're fine with going with `x/crypto/ssh`'s order, I've swapped to that and updated comments appropriately, as they prefer GCM first.